File: src/main/java/com/jhl/mds/dto/TableFieldsMappingRequestDTO.java
package com.jhl.mds.dto;

import com.jhl.mds.jsclientgenerator.JsClientDTO;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
@JsClientDTO(fileName = "table-fields-mapping-request-dto", className = "TableFieldsMappingRequestDTO")
public class TableFieldsMappingRequestDTO {

    private int sourceServerId;
    private String sourceDatabase;
    private String sourceTable;
    private int targetServerId;
    private String targetDatabase;
    private String targetTable;

    private List<SimpleFieldMappingDTO> mapping;
}

File: src/main/resources/static/resources/js/source/dto/table-fields-mapping-request-dto.js
// @flow
import {SimpleFieldMappingDTO} from './simple-field-mapping-dto';
import {observable} from 'mobx';

export class TableFieldsMappingRequestDTO {

    /**
     * @type {number}
     */
    @observable sourceServerId: ?number = 0;
    /**
     * @type {string}
     */
    @observable sourceDatabase: ?string = '';
    /**
     * @type {string}
     */
    @observable sourceTable: ?string = '';
    /**
     * @type {number}
     */
    @observable targetServerId: ?number = 0;
    /**
     * @type {string}
     */
    @observable targetDatabase: ?string = '';
    /**
     * @type {string}
     */
    @observable targetTable: ?string = '';
    /**
     * @type {SimpleFieldMappingDTO[]}
     */
    @observable mapping: ?SimpleFieldMappingDTO[] = null;

    /**
     * @param sourceServerId {number}
     * @param sourceDatabase {string}
     * @param sourceTable {string}
     * @param targetServerId {number}
     * @param targetDatabase {string}
     * @param targetTable {string}
     * @param mapping {SimpleFieldMappingDTO[]}
     */
    constructor(sourceServerId: ?number, sourceDatabase: ?string, sourceTable: ?string, targetServerId: ?number, targetDatabase: ?string, targetTable: ?string, mapping: ?SimpleFieldMappingDTO[]) {
        this.sourceServerId = sourceServerId;
        this.sourceDatabase = sourceDatabase;
        this.sourceTable = sourceTable;
        this.targetServerId = targetServerId;
        this.targetDatabase = targetDatabase;
        this.targetTable = targetTable;
        this.mapping = mapping;
    }


}
